node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -128,117 +130,52 @@ func (r OsDeviceConnectivityHelperScsiGeneric) RescanDevices(lunId int, arrayIde
-		}
-	}
-	if connectivityType == ConnectionTypeISCSI {
-		subsystemPrefix = "ip*-"
-	}
-
-	var targetPath = fmt.Sprintf("/dev/disk/by-path/%s", subsystemPrefix)
+func (r OsDeviceConnectivityHelperScsiGeneric) GetMpathDevice(volumeId string) (string, error) {
+	logger.Infof("GetMpathDevice: Searching multipath devices for volume : [%s] ", volumeId)
 
-	logger.Debugf("GetMpathDevice: Start concurrent multipath devices search for volume : [%s]", volumeId)
-	mpathResChannel := make(chan *WaitForMpathResult)
-	for _, arrayIdentifier := range arrayIdentifiers {
-		go r.waitForMpath(targetPath, connectivityType, arrayIdentifier, lunId, volumeId, mpathResChannel)
-	}
+	volumeUuid := strings.Split(volumeId, ":")[1]
+	volumeUuidLower := strings.ToLower(volumeUuid)
 
-	for i := 1; i <= len(arrayIdentifiers); i++ {
-		mpathRes := <-mpathResChannel
-		devicePaths = append(devicePaths, mpathRes.devicesPaths...)
-		if mpathRes.err != nil {
-			errStrings = append(errStrings, mpathRes.err.Error())
-		}
-	}
-	close(mpathResChannel)
-	logger.Debugf("GetMpathDevice: Finished concurrent multipath devices search for volume : [%s]", volumeId)
+	mpathdOutput, err := r.Helper.WaitForDmToExist(volumeUuidLower, WaitForMpathRetries, WaitForMpathWaitIntervalSec)
 
-	if len(devicePaths) == 0 && len(errStrings) != 0 {
-		err := errors.New(strings.Join(errStrings, GetMpahDevErrorsSep))
+	if err != nil {
 		return "", err
 	}
-
-	devicePathTosysfs := make(map[string]bool)
-	// Looping over the physical devices of the volume - /dev/sdX and store all the dm devices inside map.
-	for _, path := range devicePaths {
-		if path != "" { // since it may return empty items
-			mappedDevicePath, err := r.Helper.GetMultipathDisk(path)
-			if err != nil {
-				return "", err
-			}
-
-			if mappedDevicePath != "" {
-				devicePathTosysfs[mappedDevicePath] = true // map it in order to save uniq dm devices
-			}
-
-		}
+	if mpathdOutput == "" {
+		return "", &MultipathDeviceNotFoundForVolumeError{volumeId}
 	}
-
-	var mps string
-	for key := range devicePathTosysfs {
-		mps += ", " + key
+	dms := make(map[string]bool)
+	scanner := bufio.NewScanner(strings.NewReader(mpathdOutput))
+	for scanner.Scan() {
+		deviceLine := scanner.Text()
+		lineParts := strings.Split(deviceLine, MpathdSeparator)
+		dm, uuid := lineParts[0], lineParts[1]
+		if strings.Contains(uuid, volumeUuidLower) {
+			dmPath := filepath.Join(DevPath, filepath.Base(dm))
+			dms[dmPath] = true
+			logger.Infof("GetMpathDevice: DM found: %s for volume %s", dmPath, uuid)
+		}
 	}
-	logger.Infof("GetMpathDevice: Found multipath devices: [%s] that relats to lunId=%d and arrayIdentifiers=%s", mps, lunId, arrayIdentifiers)
 
-	if len(devicePathTosysfs) > 1 {
-		return "", &MultipleDmDevicesError{volumeId, lunId, arrayIdentifiers, devicePathTosysfs}
+	if len(dms) > 1 {
+		return "", &MultipleDmDevicesError{volumeId, dms}
 	}
 
 	var md string
-	for md = range devicePathTosysfs {
+	for md = range dms {
```

Comment:
could you rename `md` to `dm` while you're at it?

Reply:
done